.github/workflows/ci.yml

```diff
@@ -0,0 +1,50 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+      - release-*
+  pull_request: {}
+  workflow_dispatch: {}
+
+env:
+  GO_VERSION: '1.17'
```

Comment:
Note: bumped `go.mod` to `1.17` to stay aligned with our `GO_VERSION` here for clarity even though not strictly required.